### DISCUSSION TITLE
Sketchware rendering project files

### DISCUSSION THREAD

**Discussion by abrihamdemisse** [Sketchware-Pro/Sketchware-Pro#799] [Q&A]
Hello friends, please, please help me 🙏🏻. I am developing a big production app, and everything was going well until my phone ran out of storage, and suddenly everything was gone 😔 by itself and i don't have a backed up file(.swb) . However, I managed to recover the logic Java files and resources of all activities, but I couldn't recover the view. When I open an activity, all the logic and components are present, but in the view panel, there is nothing on the design. I have the full "view" file.

Let me tell you more about how I recovered it. After I saw everything was just gone, I backed up the project without saving the changes. Then I created a new project, and in the project data folder, I replaced all the files with the files from the backup file. But the "file" called file was 0B, so I remembered the name of every activity and custom view and created new ones with the same name. Then, all the logic, components, and views recovered, but some activity view designs became blank. I tried to do the same thing, but nothing worked. "AND REMEMBER THIS HAPPENED BEFORE, IN MY PREVIOUS PROJECT, but I managed to recover it fully by using this method."

So, I have the whole "view" file, but I can't recover it. If you can edit or read it, please help me. How can I make this file workable again? I can't understand how I can read this file. It isn't JSON, txt... that can be read. If you have a good knowledge of how Sketchware renders this file, please help me. I need your help.

the view file \/

[view](https://www.mediafire.com/file/9au2pu3pmz8u18c/view/file)

**Comment by PranavPurwar** [comment]
@AbrishPhoenix the file is encrypted, decrypt it with the password `sketchwaresecure` to obtain the json

**Reply by abrihamdemisse** [reply]
how , where ?

**Comment by abrihamdemisse** [comment]
with the help of @PranavPurwar answer i managed to decrypt the file and get at least the view json file, but it is not working, so , if you know how widget added in to collection tell me how so i can arrange the text and add it in list text of widget collection so that i can use the view normally and add the view in responding Activity. 

or if you can decrypt the file and look for something that made it impossible to be read if you got it fix it or tell me where the error is in decrypted view file
